src/devices/bus/a2bus/a2superdrive.cpp

```diff
@@ -0,0 +1,349 @@
+	a2bus_superdrive_device(mconfig, A2BUS_SUPERDRIVE, tag, owner, clock)
+{ }
+
+void a2bus_superdrive_device::device_start()
+{
+	save_item(NAME(m_bank_select));
+	save_item(NAME(m_side));
+}
+
+
+void a2bus_superdrive_device::device_reset()
+{
+	m_bank_select = 0;
+	m_side = 0;
+}
+
+
+// overrides of standard a2bus slot functions
+uint8_t a2bus_superdrive_device::read_c0nx(uint8_t offset)
+{
+	if (machine().side_effects_disabled()) return 0;
+
+	m_bank_select = offset & 0x0f;
+	return 0;
+}
+
+void a2bus_superdrive_device::write_c0nx(uint8_t offset, uint8_t data)
+{
+	m_bank_select = offset & 0x0f;
+}
+
+uint8_t a2bus_superdrive_device::read_cnxx(uint8_t offset)
+{
+	return m_ram[0x7b00 + offset];
+}
+
+void a2bus_superdrive_device::write_cnxx(uint8_t offset, uint8_t data)
+{
+	//m_ram[0x7b00 + offset] = data;
+}
+
+/*
+ * $c800 - $cbff is uc RAM, controlled via c0nx
+ * $cc00 - $cfff is uc RAM, hardcoded to $7c00 - $7fff
+ */
+uint8_t a2bus_superdrive_device::read_c800(uint16_t offset)
+{
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	return m_ram[address];
+}
+
+void a2bus_superdrive_device::write_c800(uint16_t offset, uint8_t data)
+{
+	unsigned address;
+
+	if (offset < 0x400)
+		address = (m_bank_select << 10) + offset;
+	else
+		address = 0x7c00 + offset - 0x400;
+
+	m_ram[address] = data;
+}
+
+
+/* uc 65c02 i/o at $0a00 */
+void a2bus_superdrive_device::m65c02_w(offs_t offset, uint8_t value)
+{
+	// $00-$0f = swim registers
+	// $40 = head sel low
+	// $41 = head sel high
+	// $80 = diagnostic led on
+	// $81 = diagnostic led off
+
+	floppy_image_device *floppy = nullptr;
+
+	if (offset < 16)
+	{
+		m_fdc->write(offset, value);
+		return;
+	}
+
+	switch (offset)
+	{
+		case 0x40:
+			m_side = 0;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x41:
+			m_side = 1;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x80:
+			logerror("LED on");
+			break;
+
+		case 0x81:
+			logerror("LED off");
+			break;
+
+		default:
+			logerror("write($0a%02x,%02x)\n", offset, value);
+			break;
+	}
+}
+
+uint8_t a2bus_superdrive_device::m65c02_r(offs_t offset)
+{
+	floppy_image_device *floppy = nullptr;
+
+	if (offset < 16)
+		return m_fdc->read(offset);
+
+	switch (offset)
+	{
+		case 0x40:
+			if (machine().side_effects_disabled()) break;
+			m_side = 0;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x41:
+			if (machine().side_effects_disabled()) break;
+			m_side = 1;
+			floppy = m_fdc->get_floppy();
+			if (floppy) floppy->ss_w(m_side);
+			break;
+
+		case 0x80:
+			logerror("LED on");
+			break;
+
+		case 0x81:
+			logerror("LED off");
+			break;
+
+		default:
+			logerror("read($0a%02x)\n", offset);
+			break;
+	}
```

Comment:
Can you lift the `if (machine().side_effects_disabled())` above this `switch` since all the things inside it have side effects?  As it is now, the LED commands and catch-all will spam when memory windows are open.